mirdata/datasets/irmas.py

```diff
@@ -0,0 +1,251 @@
+
+        # TRAINING TRACKS
+        if '__' in track_id:
+            self.predominant_instrument = os.path.basename(os.path.dirname(self.audio_path))
+            assert self.predominant_instrument in INST_DICT, "Instrument {} not in instrument dict".format(self.predominant_instrument)
+
+            # Drum presence annotation is present
+            if 'dru' in self._track_paths['audio'][0] or 'nod' in self._track_paths['audio'][0]:
+                self.genre = self._track_paths['audio'][0].split('.')[0].split('[')[3].split(']')[0]
+                self.drum = [True if 'dru' in self._track_paths['audio'][0] else False][0]
+
+            # Drum presence annotation not present
+            else:
+                self.genre = self._track_paths['audio'][0].split('.')[0].split('[')[2].split(']')[0]
+                self.drum = None
+
+        # TESTING TRACKS
+        else:
+            self.train = False
+
+    @utils.cached_property
+    def instrument(self):
+        """(list, string): predominant instrument"""
+        if self.predominant_instrument is not None:
+            return [self.predominant_instrument]
+        else:
+            return load_pred_inst(self.annotation_path)
+
+    @property
+    def audio(self):
+        """(np.ndarray, float): audio signal, sample rate"""
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            metadata={
+                'instrument': self.instrument,
+                'genre': self.genre,
+                'drum': self.drum,
+                'train': self.train
+            },
+        )
+
+
+def load_audio(audio_path):
+    """Load a IRMAS dataset audio file.
+
+    Args:
+        audio_path (str): path to audio file
+
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=44100, mono=False)
+
+
+def load_pred_inst(annotation_path):
+    """Load predominant instrument of track
+
+    Args:
+        annotation_path (str): Local path where the testing annotation is stored.
+    Returns:
+        pred_inst (str): testing track predominant instrument(s) annotations
+    """
+    if annotation_path is None:
+        return None
+
+    if not os.path.exists(annotation_path):
+        raise IOError("annotation_path {} does not exist".format(annotation_path))
+
+    pred_inst = []
+    with open(annotation_path, 'r') as fopen:
+        pred_inst_file = fopen.readlines()
+        for inst_ in pred_inst_file:
+            inst_code = inst_[:3]
+            assert inst_code in INST_DICT, "Instrument {} not in instrument dictionary".format(inst_code)
+            pred_inst.append(inst_code)
+
+        return pred_inst
```

Comment:
maybe I don't understand the dataset well - I'd expect "predominant instrument" to have only one option. Are these annotations possibly giving all the instruments, not just the predominant?

Asking to know if the labels for `predominant_instrument` in the training set have the same properties/definition as the ones in the test set. If they're different, maybe we could just rename this function.

cc @nkundiushuti

Reply:
The instrument annotations contain just the predominant instrument for both training and testing. There are no annotations about the whole instrument line-up of the tracks. 
However, the training set annotation contain just a single predominant instrument, while the testing set annotations contain more that one predominant instrument, if applicable. For instance, in a piece where the guitar and the voice are similarly predominant, the testing set annotation will contain both instruments.